src/Npgsql/PostgresException.cs

```diff
@@ -236,6 +236,40 @@ internal PostgresException(ReadBuffer buf)
         public override string Message => SqlState + ": " + MessageText;
 
         /// <summary>
+        /// Gets whether or not the exception can be treated as transient based on the SqlState. Useful when defining retry policies.
+        /// </summary>
+        public override bool IsTransient
+        {
+            get
+            {
+                switch (SqlState)
+                {   
+                    case "53000":   //insufficient_resources
+                    case "53100":   //disk_full
+                    case "53200":   //out_of_memory
+                    case "53300":   //too_many_connections
+                    case "53400":   //configuration_limit_exceeded
+                    case "57P03":   //cannot_connect_now
+                    case "58000":   //system_error
+                    case "58030":   //io_error
+                    case "40001":   //serialization_error
+                    case "55P03":   //lock_not_available
```

Comment:
I'm not convinced that the 55* errors should really be treated as transient (or at least not all of them)... At least something like "lock not available" sounds like maybe the programmer forgot to the acquire a lock? And "object in use" sounds like there's a lack of synchronization in their code...

Can you provide a bit more info on these specific ones?